PerimeterXModule/Internals/Helpers/PxCommonUtils.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Net;
+using System.Web;
+
+namespace PerimeterX
+{
+	class PxCommonUtils
+	{
+		/**
+		 * <summary>
+		 * Request helper, extracting the ip from the request according to sockerIpHeader or from 
+		 * the request socker when sockerIpHeader is absent
+		 * </summary>
+		 * <param name="context" >HttpContext</param>
+		 * <param name="pxConfig">PxConfiguration</param>
+		 * <returns>Ip from the request</returns>
+		 */
+		public static string GetRequestIP(HttpContext context, PxModuleConfigurationSection pxConfig)
```

Comment:
In theory you could only pass `pxConfig.SocketIpHeader` as an argument, but leave as is if it feels to you "more generic" the way it is.

Reply:
👍